doc/specs/#148 - Repair Support.md

```diff
@@ -0,0 +1,113 @@
+| **--moniker** | Limits the repair to the specified moniker of the application.|
+| **-v --version** | Limits the repair to the specified version of the application.If not specified, the repair will be applied to the latest version of the application.|
+| **--architectures** | Select the architecture |
+| **-s --source** | Limits the search to specified source(s).Must be followed by the source name.|
+| **-o, --log** | Directs the logging to a log file. You must provide a path to a file that you have the write rights to. |
+| **-i --interactive** | Runs the repair in interactive mode.|
+| **-h --silent** | Runs the repair in silent mode.|
+| **-?, --help** | Get additional help on this command. |
+| **--accept-source-agreements** | Accept all source agreements during source operations |
+| **--logs, --open-logs** | Open the default logs location. |
+| **--locale** | Sets the locale.|
+
+
+### Repair Feature for different Installer Types with Native Repair Capabilities
+  - Msi, Wix : The repair command will use the built-in repair features of the MSI installer type. It will run the msiexec command with the default [repair options](https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/msiexec#repair-options) to repair the application using a ShellExecute call or by invoking [MsiReinstallProduct](https://learn.microsoft.com/en-us/windows/win32/api/msi/nf-msi-msireinstallproductw) API call with `REINSTALLMODE` mode property set.
+  - Msix : The repair command will use the built-in repair features of the MSIX installer type. It will make an MSIX API call to register the application package, which will attempt to repair the package.
+    > We can't do the same thing as setting app repair for MSIX app right now because we can't use the private APIs that it uses to do repair operations with winget.
+  - MSStore : The repair command will make an MSStore API  [StartProductInstallAsync](https://learn.microsoft.com/en-us/uwp/api/windows.applicationmodel.store.preview.installcontrol.appinstallmanager.startproductinstallasync?view=winrt-22621) call to repair the application with 'Repair' property of AppInstallOption set to true.
+
+### Repair Feature for Installer Types that require Custom Repair Switch
+  - Burn, Exe, Nullsoft & Inno : The custom switch for repair in the YAML manifest file will be used to perform the repair. The repair command will run the installer with the custom switch to repair the application. To have enough flexibility, different options are possible depending on the installer source used for repair. 
+   - Installed Source: 
+     - If the YAML manifest file specifies the `Repair` switch and `Modify` as the `RepairBehavior`, the repair command will use the modify command in the ARP `ModifyPath` registry key, along with the repair switch, through a ShellExecute call, as long as `NoModify` and `NoRepair` ARP registry flags are not set to 1.
```

Comment:
To confirm, installers like the [GDK](https://github.com/microsoft/winget-pkgs/tree/master/manifests/m/Microsoft/GDK), that have modify disabled, should automatically do the right thing? 

Also am I correct that for now straight "modify" is not being supported, as in adding/removing functionality. We are just using modify if we need to trigger it for repair?

Reply:
**Answer for the first question:**
We intend to model the winget repair feature after the modify (repair) flow in the settings app. I think the settings app will either have no option to modify or the option will be disabled for the installers that do not support modify. Therefore, I expect that the 'NoModify' ARP will be set in this scenario and winget will not be able to repair from the modify flow.

The probable solution for the installers that do not support modify but support repair would be to set 'RepairBehavior' as 'Installer' and pass in the 'Repair' switch. In this case, winget will download the matching installer from the remote source and execute it with the repair switch. Is this an acceptable solution for GDK?

**Answer for the second question:**
You are correct, we are not supporting direct "modify". For the cases where 'Modify' is supported, we plan to trigger the modify flow by executing the installer that 'ModifyPath' ARP points to with the repair switch appended to it to activate the repair functionality.

Reply:
Clarification on the first question; add/remove apps has modify and uninstall. The modify UX just does not have modify. 
![image](https://github.com/microsoft/winget-cli/assets/1520739/92497494-08b3-4a09-8f44-cf4db6bb8c5d)